### PR TITLE
Add docs widget alert.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

File: docs/alert.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    ->class('alert alert-primary alert-dismissible fade show')
+    ->iconClass('bi bi-exclamation-triangle-fill flex-shrink-0 me-2')
+    ->id('w0-alert')
+    ->layoutBody('{icon}{body}{button}')
+    ->render(),
+```
+
+The code above generates the following HTML:
+```html
+<div id="w0-alert" class="alert alert-primary alert-dismissible fade show" role="alert">
```
Comment (devanych) [inline docs/alert.md:38]: <div id="w0-alert" class="alert alert-primary alert-dismissible fade show" role="alert">
    <div class="align-items-center d-flex">
        <div><i class="bi bi-exclamation-triangle-fill flex-shrink-0 me-2"></i></div>
            <span>An example alert with an icon</span>
           <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
    </div>
</div>

File: docs/alert.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    ->class('notification is-danger')
+    ->id('w0-alert')
+    ->layoutBody('{body}{button}')
+    ->render(),
+```
+
+The code above generates the following HTML:
+
+```html
+<div id="w0-alert" class="notification is-danger" role="alert">
```
Comment (devanych) [inline docs/alert.md:64]: <div id="w0-alert" class="notification is-danger" role="alert">
    <span>An example alert.</span>
    <button type="button" class="delete">&times;</button>
</div>

File: docs/alert.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    ->buttonOnClick('closeAlert()')
+    ->class('bg-red-100 border border-red-400 text-red-700 px-4 py-3 rounded relative')
+    ->id('w0-alert')
+    ->render(),
+```
+
+The code above generates the following HTML:
+
+```html
+<div id="w0-alert" class="bg-red-100 border border-red-400 text-red-700 px-4 py-3 rounded relative" role="alert">
```
Comment (devanych) [inline docs/alert.md:88]: <div id="w0-alert" class="bg-red-100 border border-red-400 text-red-700 px-4 py-3 rounded relative" role="alert">
    <span class="align-middle inline-block mr-8"><b>Holy smokes!</b> Something seriously bad happened.</span>
    <button type="button" class="absolute bottom-0 px-4 py-3 right-0 top-0" onclick="closeAlert()">&times;</button>
</div>

**Review by devanych — APPROVED** [review]

File: docs/alert.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+The code above generates the following HTML:
+```html
+<div id="w0-alert" class="alert alert-primary alert-dismissible fade show" role="alert">
+<div class="align-items-center d-flex">
+<div><i class="bi bi-exclamation-triangle-fill flex-shrink-0 me-2"></i></div>
+<span>An example alert with an icon</span>
+<button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
+</div>
+</div>
```
Comment (devanych) [inline docs/alert.md:44]:
```html
<div id="w0-alert" class="alert alert-primary alert-dismissible fade show" role="alert">
    <div class="align-items-center d-flex">
        <div><i class="bi bi-exclamation-triangle-fill flex-shrink-0 me-2"></i></div>
            <span>An example alert with an icon</span>
           <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
    </div>
</div>
```

File: docs/alert.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    ->render(),
+```
+
+The code above generates the following HTML:
+
+```html
+<div id="w0-alert" class="notification is-danger" role="alert">
+<span>An example alert.</span>
+<button type="button" class="delete">&times;</button>
+</div>
```
Comment (devanych) [inline docs/alert.md:67]:
```html
<div id="w0-alert" class="notification is-danger" role="alert">
    <span>An example alert.</span>
    <button type="button" class="delete">&times;</button>
</div>
```

File: docs/alert.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    ->render(),
+```
+
+The code above generates the following HTML:
+
+```html
+<div id="w0-alert" class="bg-red-100 border border-red-400 text-red-700 px-4 py-3 rounded relative" role="alert">
+<span class="align-middle inline-block mr-8"><b>Holy smokes!</b> Something seriously bad happened.</span>
+<button type="button" class="absolute bottom-0 px-4 py-3 right-0 top-0" onclick="closeAlert()">&times;</button>
+</div>
```
Comment (devanych) [inline docs/alert.md:91]:
```html
<div id="w0-alert" class="bg-red-100 border border-red-400 text-red-700 px-4 py-3 rounded relative" role="alert">
    <span class="align-middle inline-block mr-8"><b>Holy smokes!</b> Something seriously bad happened.</span>
    <button type="button" class="absolute bottom-0 px-4 py-3 right-0 top-0" onclick="closeAlert()">&times;</button>
</div>
```

**Review by devanych — APPROVED** [review]

File: docs/alert.md (hunk 6 above)
Comment (terabytesoftw) [reply docs/alert.md:91]: done

File: docs/alert.md (hunk 5 above)
Comment (terabytesoftw) [reply docs/alert.md:67]: done.

File: docs/alert.md (hunk 4 above)
Comment (terabytesoftw) [reply docs/alert.md:44]: done.

File: docs/alert.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+```
+
+For other examples of different Alert designs you can see the different examples in the [Alert Test](https://github.com/yiisoft/yii-widgets/blob/master/tests/AlertTest.php)
```
Comment (samdark) [inline docs/alert.md:94]:
```suggestion
For other examples of different Alert designs you can see [AlertTest](https://github.com/yiisoft/yii-widgets/blob/master/tests/AlertTest.php)
```

File: docs/alert.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+## Docs frameworks CSS
+
+- [Bootstrap5](https://getbootstrap.com/docs/5.0/components/alerts/)
+- [Bulma](https://bulma.io/documentation/elements/notification/)
+- [Tailwind](https://tailwindui.com/components/application-ui/feedback/alerts)
+
+## Assets
+
+To use this widget you must register the assets, you can see the documentation at [yiisoft/assets](https://github.com/yiisoft/assets#general-usage) our asset registration package, or just add an html link for asset registration, the package does not provide any `JS` code to use the close button of bulma and talwind.
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply docs/alert.md:13]: If it is correct to be agnostic we do not provide the asset the user must install it.

File: docs/alert.md (hunk 7 above)
Comment (terabytesoftw) [reply docs/alert.md:94]: done.

File: docs/alert.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
+# Alert widget
+
+Provide contextual feedback messages for typical user actions with the handful of available and flexible alert messages.
+
+## Docs frameworks CSS
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply docs/alert.md:5]: done.

File: docs/alert.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
+# Alert widget
+
+Provide contextual feedback messages for typical user actions with the handful of available and flexible alert messages.
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply docs/alert.md:3]: done.

File: docs/alert.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+
+## CSS frameworks documentation
+
+- [Bootstrap5](https://getbootstrap.com/docs/5.0/components/alerts/)
+- [Bulma](https://bulma.io/documentation/elements/notification/)
+- [Tailwind](https://tailwindui.com/components/application-ui/feedback/alerts)
+
+## Assets
+
+To use this widget you must register the assets, you can see the documentation at [yiisoft/assets](https://github.com/yiisoft/assets#general-usage) our asset registration package, or just add an html link for asset registration, the package does not provide any `JS` code to use the close button of bulma and talwind.
```
Comment (samdark) [inline docs/alert.md:13]:
```suggestion
This widget is meant to be used with your CSS framework of choice. The framework could be added either by [registering its assets](https://github.com/yiisoft/assets#general-usage), or by adding it directly via HTML `link` and `script` tags. The package does not provide any JavaScript code for the close button as well.
```

**Comment by samdark** [comment]
👍

**Pull request merged**